src/_pytest/assertion/util.py

```diff
@@ -96,6 +100,23 @@ def _format_lines(lines: Sequence[str]) -> List[str]:
     return result
 
 
+def _get_number_of_calls(func_name: str) -> int:
```

Comment:
While this solution is elegant in a way, I don't think using `inspect.stack()` is a good idea. It is a sort-of meta/introspection facility with it its own rules that I think is better to leave for special cases. In this case it should be possible to implement entirely "within" the language.